src/jsonpath/grammer.pest

```diff
@@ -44,20 +45,22 @@ term = _{select_term | decimal | string | boolean}
 single_filter = {term ~ op ~ term | term}
 
 filter_relation = _{and | or}
-    and = {"&&"}
-    or  = {"||"}
+    and = @{"&&"}
+    or  = @{"||"}
 
 inner_filter = _{single_filter | "(" ~ filter ~ ")"}
 
 filter = {inner_filter ~ (filter_relation ~ inner_filter)*}
 
-all = {"*"}
+all = @{"*"}
 
-full_scan = {".."}
+full_scan = @{".."}
 
-bracket = _{numbers_range | numbers_list | string_list | all | "?" ~ filter}
+square_bracket = _{"[" ~ (numbers_range | numbers_list | string_list | all | "?" ~ filter) ~ "]"}
 
-element = _{"." ~ literal | full_scan ~ (literal | all | "[" ~ bracket ~ "]") | "." ~ all | ("."? ~ "[" ~ bracket ~ "]")}
+element = _{(full_scan | ".") ~ (literal | all | square_bracket) | square_bracket}
+
+function = _{literal ~ "(" ~ (literal)? ~ ("," ~ literal)* ~ ")"}
```

Comment:
Where do we use this? Or is it a preparation for functions support?

Reply:
my mistake, I forgot it in the code for future support